relay-server/src/actors/project_buffer.rs

```diff
@@ -296,6 +296,16 @@ impl BufferService {
 
                 // spawn the task to enqueue the entire buffer
                 tokio::spawn(async move {
+                    // A RESERVED lock will be acquired when the first INSERT, UPDATE, or DELETE statement is executed.
+                    // Once we started spooling our in-memory buffer, the dequeue from the disk
+                    // will be blocked till this operations either fails, or commits.
```

Comment:
Where does it do that blocking?

Reply:
Another query, which will try to access this table will be blocked till the lock is released. This is also partially address the `fence` suggestion from yesterday. There will be definitely only 1 query at a time.